### PR TITLE
RL--based Learner for Active Face Recognition

### PR DESCRIPTION
**Pull request by Pavlos-Tosidis**
Adding Active Face Recognition agent. The implementation relies on stable_baselines3 and trains on Webots.
A Webots world is provided and the corresponding environment file to train an agent in the task of Active Face Recognition, in which the goal is to maximize FR confidence while controlling a Mavic2 drone

### CODE REVIEW

File: src/opendr/perception/active_perception/active_face_recognition/simulation/worlds/active_face_recognition.wbt [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    }
+  ]
+}
+DEF paintings_6 Transform {
+  rotation 0 0 1 3.141592653589793
+  children [
+    DEF painting_4 PortraitPainting {
+      translation 0 9.82 1.145
+      rotation 0 0 1 -1.5708
+      name "portrait painting(5)"
```
[the remark this answers is not in the document]
Comment (Pavlos-Tosidis) [reply src/opendr/perception/active_perception/active_face_recognition/simulation/worlds/active_face_recognition.wbt:174]: Changed

File: docs/reference/active_face_recognition.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+- **device**: *{'cpu', 'cuda'}, default='cuda'*\
+  Specifies the device to be used.
+- **n_steps**: *int, default=6400*\
+  Specifies the number of steps to run environment per update.
+- **gamma**: *float, default=0.9*\
+  Discount factor for future rewards.
+- **clip_range**: *float, default=0.1*\
+  Clip policy updates.
+- **target_kl**: *float, default=0.1*\
+  KL Divergence update threshold.
```
Comment (omichel) [inline docs/reference/active_face_recognition.md:39]: For consistency, we should either remove the "Specifies the" for each parameter description or add it to the last 3 descriptions. I would suggest to remove it to make it more compact.

**Review by omichel — CHANGES_REQUESTED** [review]
Thank you for this great contribution. That looks very good to me. Any reason why you are not using the latest version of Webots, e.g., R2023b?

File: docs/reference/active_face_recognition.md (hunk 2 above)
Comment (Pavlos-Tosidis) [reply docs/reference/active_face_recognition.md:39]: Even though we faced some issues with 2023b (scaled up objects on existing worlds), we do indeed use the latest version in this work as stated in the README.md (running the environment)

Comment (omichel) [reply docs/reference/active_face_recognition.md:39]: In that case, we should probably fix the proto files and wbt files as they refer to Webots R2022b or R2023a.

**Review by tsampazk — APPROVED** [review]
Thank you!

**Review by omichel — APPROVED** [review]
Very good. Thank you.

**Pull request merged**